### PR TITLE
chore(docs): document agent api debug endpoints

### PR DESCRIPTION
**Pull request by DanielleMaywood**
relates to https://github.com/coder/coder/issues/14362

For now this is just handwritten (so no Swagger involved) documentation. It covers the four agent debug endpoints: `/debug/logs`, `/debug/magicsock`, `/debug/magicsock/debug-logging/{state}` and `/debug/manifest`.

Future work should aim to include _how_ and _when_ to use the API.

### CODE REVIEW

**Comment by DanielleMaywood** [comment]
I have read the CLA Document and I hereby sign the CLA

File: docs/reference/agent-api/debug.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+`GET /debug/manifest`
+
+Get the manifest the agent fetched from `coderd` upon startup.
+
+### Responses
+
+| Status | Meaning                                                 | Description | Schema                                          |
+| ------ | ------------------------------------------------------- | ----------- | ----------------------------------------------- |
+| 200    | [OK](https://tools.ietf.org/html/rfc7231#section-6.3.1) | OK          | [agentsdk.Manifest](./schemas#agentsdkmanifest) |
```
Comment (johnstcn) [inline docs/reference/agent-api/debug.md:76]:
```suggestion
| 200    | [OK](https://tools.ietf.org/html/rfc7231#section-6.3.1) | OK          | [agentsdk.Manifest](./schemas.md#agentsdkmanifest) |
```

File: docs/reference/agent-api/schemas.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+| `owner_name`                 | string                                                                                            | true     |              |             |
+| `workspace_id`               | string                                                                                            | true     |              |             |
+| `workspace_name`             | string                                                                                            | true     |              |             |
+| `git_auth_configs`           | int                                                                                               | true     |              |             |
+| `vscode_port_proxy_uri`      | string                                                                                            | true     |              |             |
+| `apps`                       | array of [codersdk.WorkspaceApp](../api/schemas#codersdkworkspaceapp)                             | true     |              |             |
```
Comment (johnstcn) [inline docs/reference/agent-api/schemas.md:94]:
```suggestion
| `apps`                       | array of [codersdk.WorkspaceApp](../api/schemas.md#codersdkworkspaceapp)                             | true     |              |             |
```

File: docs/reference/agent-api/schemas.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+| `environment_variables`      | object                                                                                            | true     |              |             |
+| `directory`                  | string                                                                                            | true     |              |             |
+| `motd_file`                  | string                                                                                            | true     |              |             |
+| `disable_direct_connections` | boolean                                                                                           | true     |              |             |
+| `metadata`                   | array of [codersdk.WorkspaceAgentMetadataDescription](#codersdkworkspaceagentmetadatadescription) | true     |              |             |
+| `scripts`                    | array of [codersdk.WorkspaceAgentScript](../api/schemas#codersdkworkspaceagentscript)             | true     |              |             |
```
Comment (johnstcn) [inline docs/reference/agent-api/schemas.md:102]:
```suggestion
| `scripts`                    | array of [codersdk.WorkspaceAgentScript](../api/schemas.md#codersdkworkspaceagentscript)             | true     |              |             |
```

**Review by johnstcn — COMMENTED** [review]
I think the markdown link checker needs the full filename for local paths.

File: docs/reference/agent-api/debug.md (hunk 1 above)
Comment (matifali) [reply docs/reference/agent-api/debug.md:76]: .md is needed to make out link checker happy.

File: docs/reference/agent-api/debug.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
+# Debug
+
+## Get debug logs
+
+### Code samples
+
+```shell
+curl http://127.0.0.1:2113/debug/logs
```
Comment (dannykopping) [inline docs/reference/agent-api/debug.md:8]: Worthwhile referring to the config flag which defines this address (`CODER_AGENT_DEBUG_ADDRESS`), I think.

**Review by dannykopping — APPROVED** [review]
LGTM

> For now this is just handwritten (so no Swagger involved) documentation.
It would be helpful if you clarified what exactly was manually written, if not all of it. 

One other thing to check is that this won't be overwritten by running `make coderd/apidoc/swagger.json` and other targets which produce docs.

**Review by johnstcn — APPROVED** [review]
Nice work! 👍

File: docs/reference/agent-api/debug.md (hunk 4 above)
Comment (DanielleMaywood) [reply docs/reference/agent-api/debug.md:8]: Sounds good, I can make this change

**Comment by DanielleMaywood** [comment]
> One other thing to check is that this won't be overwritten by running `make coderd/apidoc/swagger.json` and other targets which produce docs.

I just ran `make coderd/apidoc/swagger.json` followed by a `git diff` and there is no diff produced and by manual inspection appears nothing has changed so I think we're good.

**Pull request merged**